tests/test_tools.py

```diff
@@ -2174,5 +2175,69 @@ def test_get_tags_invalid():
         get_tags(coll)
         assert e_info == 'ERROR: valid tags are comma or space separated strings of tag names'
 
+@pytest.mark.parametrize(
+    "inputs, expected", [
+        # everything is defined, return 1
+        ([{"repos": {"talk_repo": {"params": {"namespace_id": "35","initialize_with_readme": "false", "name": "repo name"},
+                         "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"},"prum_repo": {}}}], [1]),
+        # repos is not defined, return 0
+        ([{"repos": {}}], [0]),
+        # talk_repo is not defined, return 0
+        ([{"repos": {"talk_repo": {}, "prum_repo": {}}}], [0]),
+        # parameters are not defined
+        ([{"repos": {"talk_repo": {"params": {}, "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"},"prum_repo": {}}}], [0]),
+        # url is not defined
+        ([{"repos": {"talk_repo": {"params": {"namespace_id": "35","initialize_with_readme": "false", "name": "repo name"},"url": ""},"prum_repo": {}}}], [0]),
+        # namespace_id is not defined
+        ([{"repos": {"talk_repo": {"params": {"namespace_id": "", "initialize_with_readme": "false", "name": "repo name"},
+                         "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"}, "prum_repo": {}}}], [0]),
+        # name is not defined
+        ([{"repos": {"talk_repo": {"params": {"namespace_id": "", "initialize_with_readme": "false", "name": ""},
+                         "url": "https://gitlab.thebillingegroup.com/api/v4/projects/"}, "prum_repo": {}}}], [0]),
+    ]
+)
+def test_validate_repo(inputs, expected):
+    from regolith.runcontrol import DEFAULT_RC, load_rcfile
+    import copy
+    # import sys
+    rc = copy.copy(DEFAULT_RC)
+    rc._update(inputs[0])
```

Comment:
inputs don't have to be a list because you only pass one input in to the function.  You can make the code more readable by giving it a more descriptive name too in this case.  So instead of inputs, maybe call it `repo_information`.  Also, since this is a test of an abstract function, calling your repo `talk_repo` is ok, but your target audience of a future developer may be scratching her head.  Think about something like:
```
{'repos': {
    'repo1':{
       'params': {
           'namespace_id': "",
           'initialize_with_readme': 'false',
           ...
                       }
                }
             }
  "repo2": {}
}}
```
Looking at this structure, I wonder what work `'params'` is doing.  Isn't it redundant?  Does `'repo1` have anything else than `params` or will it in the future.  What can you think of that repo1 has that is not a parameter?  OK, to leave it there if there is something, but better to remove it if not.

Reply:
inputs changed to repo_information and no longer a list. 
repo1 also contains url, and params is passed as a dict in the request!